### ISSUE TITLE
Get-ITGluePasswords filters not applying

### ISSUE THREAD

**Issue by DavidSvirskis** [itglue/powershellwrapper#46]
I've tried these filters against the Get-ITGluePasswords cmdlet and they are not filtering any results.

-filter_name
-filter_password_category_id
-filter_organization_id
-filter_id

e.g.  this cmdlet should return nothing yet it returns a list of all passwords with any category ID:
(Get-ITGluePasswords -filter_password_category_id 123456789).data.attributes.'password-category-id'

**Comment by ecspresso** [comment]
`-body $body` is missing from Invoke-RestMethod in Get-ITGluePasswords.

**Comment by nedgeworth** [comment]
I'm seeing the same issue with -page_size against this.

**Comment by ecspresso** [comment]
A fix for this should be added to my pull request. New-ITGluePasswords checks for $flexible_asset_type_id instead of $organization_id when deciding on nested route. Add the organization id into the data request for now instead.

```posh
$body = @{
	type = 'passwords'
	attributes = @{	   
		organization_id = 1234567
		name = 'test password'
		username = 'root'
		password = 'RByx24fBkDiW'
	} 
}
New-ITGluePasswords -data $body
```

_(or manually update row 13 in /Resources/Passwords.ps1 from `if ($flexible_asset_type_id) {` to `if ($organization_id) {`)_

**Comment by CalebAlbers** [comment]
Thanks, @ecspresso! I ran into this issue today as well when attempting pagination.

It'll get fixed and merged this week, and an update to PowerShell Gallery will go out accordingly.

**Comment by nedgeworth** [comment]
Just to add to this I'm also experiencing issues with removing passwords. Not sure I'm using the correct format, but have tried the below. (Apologies for my terrible use of markdown)

Remove-ITGluePasswords -id 1234567
cmdlet Remove-ITGluePasswords at command pipeline position 1
Supply values for the following parameters:
data: 

Remove-ITGluePasswords -filter_id 1234567 -filter_organization_id 1234567 -data $data
Remove-ITGluePasswords : {"errors":[{"status":404,"title":"Not found","detail":"Record not found","source":{"pointer":"id"}}]}
At line:1 char:1
+ Remove-ITGluePasswords -filter_id 1234567 -filter_organization_id 123 ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Remove-ITGluePasswords

$data = @{
    type       = 'passwords'
    id         = '1234567'
    attributes = @{
      organization_id = '1234567'
    }
  }
Remove-ITGluePasswords -data $data

Remove-ITGluePasswords : {"errors":[{"status":404,"title":"Not found","detail":"Record not found","source":{"pointer":"id"}}]}
At line:1 char:1
+ Remove-ITGluePasswords -data $data
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Remove-ITGluePasswords

**Comment by ecspresso** [comment]
@CalebAlbers I pushed some more changes into #47.

- The parameter decleration was wrong in Set-ITGluePasswords and Remove-ITGluePasswords.
`[CmdletBinding(DefaultParameterSetName)]` instead of `[Parameter(ParameterSetName)]`.

- I updated the `data` variable to only be mandatory for the parameter sets `update` and `bulk_destroy` since when removing a single password only requires the ID in the URL. I also did an if/else in the try/catch to handle single removal/bulk removal (DELETE vs PATCH and not body when DELETE). Please review this and rewrite if needed to follow the standard of the rest of the code.

- Please update to correct version number if/when merging.

**Comment by adrianwells** [comment]
@DavidSvirskis, @nedgeworth, @CalebAlbers this should be fixed now with @ecspresso's PR #47 being merged into v2.0.4.  Closing this issue, if there is still an issue please re-open this issue.  Thank you.
